lib/useOnyx.ts

```diff
@@ -204,11 +238,11 @@ function useOnyx<TKey extends OnyxKey, TReturnValue = OnyxValue<TKey>>(key: TKey
             previousValueRef.current = newValueRef.current;
 
             // If the new value is `null` we default it to `undefined` to ensure the consumer gets a consistent result from the hook.
-            resultRef.current = [previousValueRef.current as CachedValue<TKey, TReturnValue>, {status: newFetchStatus ?? 'loaded'}];
+            resultRef.current = [previousValueRef.current ?? undefined, {status: newFetchStatus ?? 'loaded'}];
```

Comment:
Do you think we need to add `?? undefined`? I think at this point of the code it would never be `null`.

Reply:
Not sure from execution perspective, but the TS was complaining there about `previousValue` being possibly `null`, so just to be safe I added the `undefined` fallback. 

If you feel like we should replace it with assertion

```tsx
resultRef.current = [previousValueRef.current as NonNullable<TReturnValue> | undefined, {status: newFetchStatus ?? 'loaded'}];
```

We can swap it out